src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,195 @@
+            final Set<Task> tasks = taskRepository.findAllByAssigneeId(userId);
+            if (!tasks.isEmpty()) {
+                userRepository.findByRolesId(MANAGER_ROLE_ID)
+                        .forEach(m -> emailService.sendEmail(m.getEmail(),
+                                PUT_ASSIGNEE, getBodyTextNewAssignee(tasks, user)));
+            }
+        }
+        user.setRoles(roles);
+        final UserResponseDtoWithRole userDtoWithRole = userMapper
+                .toUserResponseWithRole(userRepository.save(user));
+        Set<String> roleDtos = getRoleDtos(roles);
+        userDtoWithRole.setRoleDtos(roleDtos);
+        emailService.sendEmail(user.getEmail(), UPDATE_ROLES, TOP_BASE_UPDATE_ROLES + roleDtos);
+        return userDtoWithRole;
+    }
+
+    @Override
+    public UserResponseDtoWithRole getProfileInfo(Long userId) {
+        final User user = userRepository.findById(userId).orElseThrow(
+                () -> new EntityNotFoundException("User with id " + userId + " is not exist"));
+        final UserResponseDtoWithRole userDtoWithRole = userMapper.toUserResponseWithRole(user);
+        userDtoWithRole.setRoleDtos(getRoleDtos(user.getRoles()));
+        return userDtoWithRole;
+    }
+
+    @Transactional
+    @Override
+    public UserResponseDtoWithRole updateProfileInfo(
+            Long userId, UserRegistrationRequestDto requestDto) {
+        final String emailRequestDto = requestDto.email();
+        final User userBefore = userRepository.findById(userId).orElseThrow(
+                () -> new EntityNotFoundException("User with id " + userId + " is not exist"));
+        if (userRepository.findByEmail(emailRequestDto).isPresent()
+                && !userBefore.getEmail().equals(emailRequestDto)) {
+            throw new RegistrationException("This email address is already registered");
+        }
+        final String usernameRequestDto = requestDto.username();
+        if (userRepository.findByUsername(usernameRequestDto).isPresent()
+                && !userBefore.getUsername().equals(usernameRequestDto)) {
+            throw new RegistrationException("Someone registered with this username");
+        }
+        userBefore.setUsername(usernameRequestDto);
+        userBefore.setPassword(passwordEncoder.encode(requestDto.password()));
+        userBefore.setEmail(emailRequestDto);
+        userBefore.setFirstName(requestDto.firstName());
+        userBefore.setLastName(requestDto.lastName());
+        final UserResponseDtoWithRole userDtoWithRole = userMapper
+                .toUserResponseWithRole(userRepository.save(userBefore));
+        userDtoWithRole.setRoleDtos(getRoleDtos(userBefore.getRoles()));
+        return userDtoWithRole;
+    }
+
+    private Set<String> getRoleDtos(Set<Role> userRoles) {
+        return userRoles.stream().map(r -> r.getName().toString()).collect(Collectors.toSet());
+    }
+
+    private String getBodyTextNewPerson(Long userId, String username, String email,
```

Comment:
```suggestion
    private String getNewPersonBodyText(Long userId, String username, String email,
```
same for next method